doc/specs/#3327 - Application Theming/#3327 - Application Theming.md

```diff
@@ -0,0 +1,500 @@
+set of properties that seems appropriate to include as part of a "v1" theming
+implementation. In [Future Considerations](#future-considerations), we'll
+enumerate additional properties that could be added in the future to further
+control the UI.
+
+#### Theming v1 Properties
+
+* `tab.cornerRadius`: Control the radius of the corners of the tab items.
+  Accepts a `double`. If this is set to `0`, then the tabs will have squared-off
+  corners. No particular limit is set on the max value accepted, though larger
+  values might not be aesthetically pleasing.
+* `tab.closeButton`: Control the visibility of the close button for a tab item.
+  Accepts the following values:
+    - `visible`: The default behavior of the tab item close button - always
+      visible.
+    - `hover`: The close button on a tab item only appears when the tab is
+      hovered.
+    - `hidden`: The close button on a tab is always hidden.
+* `tab.icon`: Control the visibility, appearance of the tab icon
+    - `visible`: The default behavior of the tab item icon - always visible, and
+      in full color.
+    - `outline`: The icon is always visible, but is only drawn as an outline,
+      using `BitmapIconSource.ShowAsMonochrome(true)`
+    - `hidden`: The icon is hidden
+* `tab.background`: Control the color of the background of tab items. See below
+  for accepted colors.
+* `tabRow.background`: Control the color of the background of the tab row items.
+  See below for accepted colors.
+* `pane.borderColor`: Control the color of the border used to separate panes.
+  This is the color of the inactive border between panes.
+* `pane.activeBorderColor`: Control the color of the border of the active pane
+* `pane.borderWidth`: Control the width of the borders used to separate panes.
+* `window.applicationTheme`: If set, will set the XAML `RequestedTheme`
+  property. This can be one of `light`, `dark` or `system`. This controls how
+  XAML fundamentally styles UI elements. If not provided, will use the default
+  value "system", which will use whatever the system's default theme is.
+
+#### Theme Colors
+
+For properties like `tab.background` and `tabRow.background`, these colors can
+be one of:
+* an `#rrggbb`, `#aarrggbb` color. (Alpha is ignored for `tabRow.background`)
+* `accent` for the accent color
+* `terminalBackground` to use the default background color of the active
+  terminal instance.
+* `terminalForeground` to use the default foreground color of the active
+  terminal instance.
+* `key:SomeXamlKey` to try and look `SomeXamlKey` up from our resources as a
+  `Color`, and use that color for the value.
+    - TODO DISCUSSION: Does anyone want this?
+    - is `accent` just `key:SystemAccentColor`? If it is, is it a reasonable
+      alias that we'd want to provide anyways?
+    - TODO DISCUSSION: PR[#5280] suggested `{ "key": "SomeResourceKey" }` for
+      string resources, should we use that format for colors like this as well?

```

Comment:
Why `key:SomeXamlKey` as opposed to `SomeXamlKey` directly? We can just add valid enums. Then extensions can add more enums too. Maybe I'm just not understanding this part.

Reply:
I'm missing the thread it was mentioned in now, but someone once suggested using `"xaml:SomeXamlKey"`, and I really like that. 

I think the "object" notation for looking up a key like this made it easier for us to say "this is an object, it's definitely not a string value". You're right though, these are just enums, so we can manually make sure that `xaml:` isn't used as a prefix for any of them.